e2e/imgbuild/imgbuild.go

```diff
@@ -1530,43 +1530,132 @@ echo 'export LEGACY_TEST_ENV=legacy-value' >> $SINGULARITY_ENVIRONMENT
 	)
 }
 
+func (c *imgBuildTests) testContainerBuildUnderFakerootModes(t *testing.T) {
+	tmpDir, cleanup := e2e.MakeTempDir(t, c.env.TestDir, "test-container-build-under-fakeroot-modes-", "")
+	defer cleanup(t)
+
+	// running under the mode 1, 1a (--with-suid) (https://apptainer.org/docs/user/main/fakeroot.html)
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.UserProfile),
+		e2e.WithCommand("build"),
+		e2e.WithArgs("--force", "--fakeroot", fmt.Sprintf("%s/openssh-mode1a.sif", tmpDir), "testdata/regressions/issue_476.def"),
+		e2e.ExpectExit(0),
+	)
+
+	// running under the mode 1, 1b (--without-suid) (https://apptainer.org/docs/user/main/fakeroot.html)
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.UserProfile),
+		e2e.WithCommand("build"),
+		e2e.WithArgs("--force", "--fakeroot", fmt.Sprintf("%s/openssh-mode1b.sif", tmpDir), "testdata/regressions/issue_476.def"),
+		e2e.WithEnv([]string{"_APPTAINER_E2E_NOT_USE_SUID=1"}),
+		e2e.ExpectExit(0),
+	)
+
+	// running under the mode 2(https://apptainer.org/docs/user/main/fakeroot.html)
+	// temporally remove executable permission from fakeroot
+	e2e.Privileged(func(t *testing.T) {
+		cmd := exec.Command("chmod", "-x", "/usr/bin/fakeroot")
+		err := cmd.Run()
+		if err != nil {
+			t.Fatal(err)
+		}
+	})(t)
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.FakerootModeTwoProfile),
```

Comment:
I think there will be no more need for a separate Profile for each mode if we're controlling the mode via environment variables.  Another way to do it would be to keep the profiles, but if that's done then I think they should set the environment variable instead of setting it here.  I don't think it makes sense to have both the special profile and the environment variable.